packages/yarnpkg-fslib/tests/patchedFs.test.ts

```diff
@@ -30,4 +30,13 @@ describe(`patchedFs`, () => {
       done();
     });
   });
+
+  it(`in case of the parameter of fs.exists is not a string, give false`, done => {
+    const patchedFs = extendFs(fs, new PosixFS(new NodeFS()));
+
+    patchedFs.exists(undefined as any, exists => {
```

Comment:
Shouldn't the sync version also have this behaviour?

Reply:
Actually, it already had it. There is a catch all exception handler that just returns false.